lib/eslint/eslint.js

```diff
@@ -97,7 +97,10 @@ const { Retrier } = require("@humanwhocodes/retry");
 const FLAT_CONFIG_FILENAMES = [
     "eslint.config.js",
     "eslint.config.mjs",
-    "eslint.config.cjs"
+    "eslint.config.cjs",
+    "eslint.config.ts",
+    "eslint.config.mts",
+    "eslint.config.cts"
```

Comment:
What about `eslint.config.tsx`? It's a legitimate file extension that some frontend-area projects enforce all files adhere to for consistency.

Reply:
Can you give some examples?

Reply:
I am not sure about it personally. To me, TSX is a DSL specific to some frontend frameworks. And if we going to have `.tsx`, one could argue to have `.jsx` - which is not supported by Node.js without transpiling. I am not sure what's the use case of having TSX in the config file, and such a conversion does not seem to widely supported by other tools.

Reply:
I think it's fine to leave out `.tsx` files for the moment since the RFC only mentions `.ts`, `.cts` and `.mts` files. Deciding whether a new format should be would require additional analysis.

Reply:
> Can you give some examples?

Of using a `.jsx` or `.tsx` extension for an ESLint config file? I don't only see searching public repositories. Which is probably good motivation to wontfix this suggestion.

> To me, TSX is a DSL specific to some frontend frameworks. And if we going to have `.tsx`, one could argue to have `.jsx` - which is not supported by Node.js without transpiling

Technically, only `.jsx` is a DSL specific to some frontend frameworks. `.tsx` is a file extension indicating first-class supported syntax within TypeScript. One cannot claim 100% TypeScript language/syntax compatibility without supporting `.tsx`.

Still, +1 to the resolution of not doing this. 👍